### PR TITLE
Feature/pwa

### PR DESCRIPTION
**Pull request by akopp90**

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+  experimental: {
+    newNextLinkBehavior: true,
+  },
+
   compiler: {
-    styledComponents: false,
+    styledComponents: true,
   },
-  transpilePackages: ["next-auth"],
   reactStrictMode: true,
```
Comment (F-Kirchhoff) [inline next.config.js:15]: Why did you change the styledComponents and next auth settings?

File: pages/_app.js [hunk 2]
```diff
@@ -186,6 +190,48 @@ export default function App({
...
+    window.addEventListener("appinstalled", () => {
+      setShowInstallPrompt(false);
+      setShowInstallButton(true);
+      console.log("App installed");
+      setDeferredPrompt(null);
+    });
+
+    return () => {
+      window.removeEventListener("beforeinstallprompt", () => {});
+      window.removeEventListener("appinstalled", () => {});
```
Comment (F-Kirchhoff) [inline pages/_app.js:220]: this does not remove the eventListeners. You need to store the callback function in a variable and reference it in the removeEventListener call:
```js
function prepareInstallation() { ...}
window.addEventListener("beforeinstallprompt", prepareInstallation)
return () => {
        window.removeEventListener("beforeinstallprompt", prepareInstallation);
}
```

File: pages/_app.js [hunk 3]
```diff
@@ -186,6 +190,48 @@ export default function App({
...
+    };
+  }, []);
+
+  const handleInstallClick = async () => {
```
Comment (F-Kirchhoff) [inline pages/_app.js:224]: your other handlers are defined as functions, please use the same syntax for the same concepts (like functions inside your components). I mentioned this now multiple times, please make sure to double check your code before opening the PR.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Andreas,
the installation logic looks alright, but there is some duplication of code and some errors handling the side effects. Please take a look.

File: next.config.js (hunk 1 above)
Comment (akopp90) [reply next.config.js:15]: Oh hatte die Next config vorbereitet bevor ich next auth integriert hatte 😅, binde ich wieder ein

**Review by Bianca-Gundel — APPROVED** [review]

**Comment by Bianca-Gundel** [comment]
LGTM

**Review by F-Kirchhoff — APPROVED** [review]
